macros/provider_test.go

```diff
@@ -0,0 +1,464 @@
+			args: args{
+				reqWrapper: &openrtb_ext.RequestWrapper{
+					BidRequest: &openrtb2.BidRequest{
+						Device: &openrtb2.Device{
+							Lmt: &lmt,
+						},
+						Ext: []byte(`{"prebid":{"channel": {"name":"test1"}}}`),
+					},
+				},
+			},
+			want: map[string]string{MacroKeyBidID: "", MacroKeyAppBundle: "", MacroKeyDomain: "", MacroKeyPubDomain: "", MacroKeyPageURL: "", MacroKeyAccountID: "", MacroKeyLmtTracking: "10", MacroKeyConsent: "", MacroKeyBidder: "", MacroKeyIntegration: "", MacroKeyVastCRTID: "", MacroKeyAuctionID: "", MacroKeyChannel: "test1", MacroKeyEventType: "", MacroKeyVastEvent: ""},
+		},
+		{
+			name: " custom macros present key",
+			args: args{
+				reqWrapper: &openrtb_ext.RequestWrapper{
+					BidRequest: &openrtb2.BidRequest{
+						Ext: []byte(`{"prebid":{"macros":{"CUSTOMMACR1":"value1"}}}`),
+					},
+				},
+			},
+			want: map[string]string{"PBS-MACRO-CUSTOMMACR1": "value1", MacroKeyBidID: "", MacroKeyAppBundle: "", MacroKeyDomain: "", MacroKeyPubDomain: "", MacroKeyPageURL: "", MacroKeyAccountID: "", MacroKeyLmtTracking: "", MacroKeyConsent: "", MacroKeyBidder: "", MacroKeyIntegration: "", MacroKeyVastCRTID: "", MacroKeyAuctionID: "", MacroKeyChannel: "", MacroKeyEventType: "", MacroKeyVastEvent: ""},
+		},
+		{
+			name: " All request macros present key",
+			args: args{
+				reqWrapper: &openrtb_ext.RequestWrapper{
+					BidRequest: &openrtb2.BidRequest{
+						ID: "123",
+						Site: &openrtb2.Site{
+							Domain: "testdomain",
+							Publisher: &openrtb2.Publisher{
+								Domain: "publishertestdomain",
+								ID:     "testpublisherID",
+							},
+							Page: "pageurltest",
+						},
+						App: &openrtb2.App{
+							Domain: "testdomain",
+							Bundle: "testbundle",
+							Publisher: &openrtb2.Publisher{
+								Domain: "publishertestdomain",
+								ID:     "testpublisherID",
+							},
+						},
+						Device: &openrtb2.Device{
+							Lmt: &lmt,
+						},
+						User: &openrtb2.User{Ext: []byte(`{"consent":"1" }`)},
+						Ext:  []byte(`{"prebid":{"channel": {"name":"test1"},"macros":{"CUSTOMMACR1":"value1"}}}`),
+					},
+				},
+			},
+			want: map[string]string{"PBS-MACRO-CUSTOMMACR1": "value1", MacroKeyBidID: "", MacroKeyAppBundle: "testbundle", MacroKeyDomain: "testdomain", MacroKeyPubDomain: "publishertestdomain", MacroKeyPageURL: "pageurltest", MacroKeyAccountID: "testpublisherID", MacroKeyLmtTracking: "10", MacroKeyConsent: "1", MacroKeyBidder: "", MacroKeyIntegration: "", MacroKeyVastCRTID: "", MacroKeyAuctionID: "123", MacroKeyChannel: "test1", MacroKeyEventType: "", MacroKeyVastEvent: ""},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			b := &macroProvider{
+				macros: map[string]string{},
+			}
+			b.populateRequestMacros(tt.args.reqWrapper)
+			output := map[string]string{}
+			for key := range tt.want {
+				output[key] = b.GetMacro(key)
+			}
+			assert.Equal(t, output, tt.want, tt.name)
```

Comment:
The order here for `assert.Equal` arguments needs to be `expected` then `actual`. So, can you please change it to:
```
assert.Equal(t, tt.want, output, tt.name)
```